src/modules/QtQml/Component.js

```diff
@@ -70,10 +70,17 @@ class QMLComponent {
     const bp = engine.$basePath;
     engine.$basePath = this.$basePath ? this.$basePath : engine.$basePath;
 
+    const context = this.$context ? Object.create(this.$context)
+      : new QMLContext();
+
+    if (this.importContextId !== undefined) {
+        context.importContextId = this.importContextId;
```

Comment:
Ah, it was amended in a follow-up commit.

I will partially squash while landing.